runtime/src/main/cpp/Arrays.cpp

```diff
@@ -32,6 +34,17 @@ ALWAYS_INLINE inline void mutabilityCheck(KConstRef thiz) {
   }
 }
 
+template<typename T>
+inline void fillImpl(KRef thiz, KInt fromIndex, KInt toIndex, T value) {
+  ArrayHeader* array = thiz->array();
+  checkRangeIndexes(fromIndex, toIndex, array->count_);
+  mutabilityCheck(thiz);
+  T* address = PrimitiveArrayAddressOfElementAt<T>(array, fromIndex);
```

Comment:
Is this operation well defined when `fromIndex == array.size` and thus address is pointing past the last array element, given that it's not dereferenced in the loop below?

Reply:
Yes, I believe so.